Formula/texmaker.rb

```diff
@@ -0,0 +1,64 @@
+class Texmaker < Formula
+  desc "Free cross-platform LaTeX editor since 2003"
+  homepage "https://www.xm1math.net/texmaker/"
+  url "https://www.xm1math.net/texmaker/texmaker-5.1.2.tar.bz2"
+  sha256 "526896f2c1ae561130eec7aae815b9dcda9e8eeb772b6541d0dc94ce91a71044"
+  license "GPL-2.0-only"
+
+  depends_on "qt"
+
+  on_linux do
+    depends_on "gcc"
+  end
+  on_macos do
+    depends_on xcode: :build
+  end
```

Comment:
`depends_on :xcode` is a no-op for Linux, so you don't need to wrap this in an `on_macos` block.